src/mapper.rs

```diff
@@ -52,3 +57,22 @@ pub trait Mapper {
     /// ```
     fn unmap(&mut self, virt_start: usize, bytes: usize);
 }
+
+/// The trivial mapper, which maps an address into itself.
+///
+/// This mapper serves two purposes:
+/// - It maps a physical address into the virtual address of the same value.
+///   This is especially useful when you have an access to the physical address space
+///   itself and should reference it directly, as when working on an OS kernel.
+/// - It maps an already-mapped virtual address into itself, preventing duplicate
+///   mapping calls to other non-trivial mappers.
+///
+/// An accessor of some type using this mapper may be regarded as a pointer to that type.
```

Comment:
Thank you for the enhanced document. I'm wondering if we should add a safety note for use of this struct.

On the one hand, this document should have the note as using it without correctly setting identity mapping can violate memory safety.

On the other hand, having this struct itself doesn't violate it. Instead, calling `Mapper::map` (which is already `unsafe`) either directly or indirectly (e.g., `new` of accessor structs, and they are also `unsafe`) without satisfying the necessary conditions or correct implementations can be the cause of the violation, and thus documents of accessor types should be updated instead of this structs' one.

What do you think?

Reply:
I don't think it should be in this PR.
The memory safety violation risk exists in every mapper you can implement. If we sufficiently specify the context in which this mapper should be used, the the documentation is enough.